x-pack/platform/plugins/shared/lens/common/references/index.ts

```diff
@@ -21,10 +21,25 @@ export const injectLensReferences = (
     return clonedState;
   }
 
-  // TODO: find a way to cull erroneous dashboard references
-  const combinedReferences = uniqBy([...references, ...clonedState.attributes.references], 'name');
+  const panelReferencesMap = references.reduce<Map<string, Reference>>((acc, reference) => {
+    const name = reference.name.split(':').at(-1); // name can be dashboard or panel reference name
 
-  clonedState.attributes.references = combinedReferences;
+    if (name) {
+      acc.set(name, {
+        ...reference,
+        name, // panel reference name
+      });
```

Comment:
I think we could defer this spreading to line 39:

```ts
const panelReferencesMap = references.reduce<Map<string, Reference>>((acc, reference) => {
    ...
    acc.set(name, reference);
  }
  return acc;
}, new Map());

const newReferences = clonedState.attributes.references.map((lensRef) => {
    const panelReference = panelReferencesMap.get(lensRef.name);
    return panelReference?.type === lensRef.type ? {...panelReference, name } : lensRef;
  });
```

Reply:
Fixed in 2f4ebd21f126499dc301cf0e57579d803f77321e